[EASI-3232] Select resolution page

aterstriep opened the pull request:
# EASI-3232

[Figma
](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=1%3A24513&mode=design&t=4vYeXeppt9vfSOSj-1)

## Changes and Description

- Created Select Resolution view (src/views/GovernanceReviewTeam/Actions/Resolutions.tsx)
  - Conditional logic for rendering title, description, and resolution options (see [Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=1%3A25685&mode=design&t=qtMtNyeCNLabcHmc-1))
  - Summary box renders only available resolutions
- Routes for each resolution
  - Issue a Life Cycle ID (/governance-review-team/:systemIntakeId/resolutions/issue-lcid)
  - Not an IT Governance request (/governance-review-team/:systemIntakeId/resolutions/not-it-request)
  - Not approved by GRB  (/governance-review-team/:systemIntakeId/resolutions/not-approved)
  - Close request  (/governance-review-team/:systemIntakeId/resolutions/close-request)
  - Re-open request  (/governance-review-team/:systemIntakeId/resolutions/re-open-request)
- Basic page layout for resolution forms
  - Added `<ActionForm>` component
  - Created "Selected resolution" box with action title and "Change resolution" link

<!-- Put a description here! -->

## How to test this change

- Make sure `itGovV2Enabled` flag is enabled
- Go to /governance-review-team/:systemIntakeId/actions
- Select "Issue a decision or close this request"
- Ensure Resolutions page works as expected

<!--
    Add any steps or code to run in this section to help others run your code:

    ```sh
    echo "Code goes here"
    ```
--->

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)


## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

patrickseguraoddball commented:
I'm a little confused by the statuses of the intakes.  I may not have context or out of the loop.  The status being calculated on the home table and also on the request are using a list of request statuses that determine a status of CLOSED within the UI.

https://github.com/CMSgov/easi-app/blob/910966057cd8012e52f9a28f4bcfeddd7a0a9aef/src/types/systemIntake.ts#L23-L30

This doesn't seem to match up with the available actions, such as to reopen a request.  Is this intended?

<img width="777" alt="Screenshot 2023-09-13 at 3 03 14 PM" src="https://github.com/CMSgov/easi-app/assets/95709965/7c0b4538-c78c-4a9a-99df-238f5380e584">
<img width="336" alt="Screenshot 2023-09-13 at 3 04 14 PM" src="https://github.com/CMSgov/easi-app/assets/95709965/826d3364-eb6d-46a6-9820-27ede0c64819">

aterstriep commented:
> This doesn't seem to match up with the available actions, such as to reopen a request. Is this intended?

[Explanation here](https://github.com/CMSgov/easi-app/pull/2175#issuecomment-1718279578)

patrickseguraoddball reviewed (changes requested):
The 'Not an IT Governance request', seems to render the selection page with only the selected resolution box above

I believe it's a mismatch/naming of routes.  The post-selection is being routed to /not-it-request, but here it's defined as 
https://github.com/CMSgov/easi-app/blob/910966057cd8012e52f9a28f4bcfeddd7a0a9aef/src/views/GovernanceReviewTeam/Actions/Resolutions.tsx#L169-L170


<img width="797" alt="Screenshot 2023-09-13 at 3 46 02 PM" src="https://github.com/CMSgov/easi-app/assets/95709965/a8119568-fe18-44b4-b08d-d679b83f59a7">

aterstriep commented:
> I believe it's a mismatch/naming of routes. The post-selection is being routed to /not-it-request, but here it's defined as

@patrickseguraoddball I fixed the route and also added a condition to show "page not found" if a user tries to access an incorrect route or unavailable action (ex: tries to close an already closed request)

patrickseguraoddball reviewed (approved):
Sweet works great, thanks!

aterstriep commented:
@patrickseguraoddball had to fix a few merge conflicts with main - can you re-approve this? Thank you!
